src/parser/shared/modules/items/bfa/raids/crucibleofstorms/TridentOfDeepOcean.js

```diff
@@ -0,0 +1,192 @@
+import React from 'react';
+
+import { calculatePrimaryStat } from 'common/stats';
+import { formatNumber, formatPercentage } from 'common/format';
+
+import ITEMS from 'common/ITEMS/index';
+import SPELLS from 'common/SPELLS/index';
+
+import Analyzer, { SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events from 'parser/core/Events';
+import ItemStatistic from 'interface/statistics/ItemStatistic';
+import ItemHealingDone from 'interface/others/ItemHealingDone';
+import BoringItemValueText from 'interface/statistics/components/BoringItemValueText';
+import StatTracker from 'parser/shared/modules/StatTracker';
+
+import PrimaryStatIcon from 'interface/icons/PrimaryStat';
+import UptimeIcon from 'interface/icons/Uptime';
+
+const MS_BUFFER = 100;
+
+// Example log: https://www.warcraftlogs.com/reports/6xHLtAFW4yC73mRD/#fight=28&source=2
+
+class TridentOfDeepOcean extends Analyzer {
+  static dependencies = {
+    statTracker: StatTracker,
+  };
+
+  static shieldDuration = 40;
+
+  damageAbsorbed = 0;
+  shieldCount = 0;
+
+  constructor(...args){
+    super(...args);
+    this.active = this.selectedCombatant.hasMainHand(ITEMS.TRIDENT_OF_DEEP_OCEAN.id);
+    if(!this.active){
+      return;
+    }
+
+    this.addEventListener(Events.absorbed.to(SELECTED_PLAYER).spell(SPELLS.CUSTODY_OF_THE_DEEP_SHIELD), this._absorbDamage);
+    this.addEventListener(Events.applybuff.to(SELECTED_PLAYER).spell(SPELLS.CUSTODY_OF_THE_DEEP_SHIELD), this._applyShield);
+    this.addEventListener(Events.refreshbuff.to(SELECTED_PLAYER).spell(SPELLS.CUSTODY_OF_THE_DEEP_SHIELD), this._applyShield);
+
+    const item = this.selectedCombatant.mainHand;
+    this.mainstatRating = calculatePrimaryStat(395, 181, item.itemLevel);
+    this.statTracker.add(SPELLS.CUSTODY_OF_THE_DEEP_BUFF.id, {
+      agility: this.mainstatRating,
+      strength: this.mainstatRating,
+      intellect: this.mainstatRating,
+    });
+  }
+
+  _applyShield(event){
+    this.shieldCount++;
+  }
+
+  isBuggedStackChange(previousTime, currentTime){
+    return previousTime > (currentTime - MS_BUFFER);
+  }
+
+  get cleanStacks() {
+    const cleanStacks = {
+      0: [
+        {
+          start: this.owner.fight.start_time,
+          end: null,
+          duration: null,
+        },
+      ],
+    };
+    let lastHandledStack = 0;
+
+    const buffStacks = this.selectedCombatant.getBuffHistory(SPELLS.CUSTODY_OF_THE_DEEP_BUFF.id).reduce((obj, buff) => {
+      obj[buff.start] = buff.stackHistory;
+      return obj;
+    }, {});
+
+    Object.values(buffStacks).forEach((stackChain) => {
+      stackChain.forEach((stack) => {
+        const stackSize = stack.stacks;
+        const stackStart = stack.timestamp;
+        if (cleanStacks[lastHandledStack]) {
+          const previousStack = cleanStacks[lastHandledStack];
+          const lastOccurrence = previousStack[previousStack.length - 1];
+          if (this.isBuggedStackChange(lastOccurrence.start, stackStart)){
```

Comment:
what exactly does "bugged" mean here? Looking at the definition of `isBuggedStackChange` it appears you're looking for stack changes that within a 100ms window? A comment explaining what this condition is would be useful

Reply:
I actually ended up using the stack tracking code from the Navigation enchant analyzer, since I looked there to see how multiple stack average uptime had been done previously. In my opinion this could use either being moved into a separate module so that it can be reused more freely or manually rewritten using the event listeners (buffapply, refresh, etc). Let me know if you want me to do either of these things.

Other than that, I believe it is just checking that the same buff didnt register as changing in too short of a delay, as this wouldn't be realistic / bugged.

Reply:
👍 i will add an issue for it and approve the PR